services/common/src/components/explosivespermits/ExplosivesPermitDiffModal.tsx

```diff
@@ -0,0 +1,257 @@
+      setCurrentDiff(differencesList);
+    }
+  }, [explosivesPermit]);
+
+  const valueOrNoData = (value: any) => {
+    if (typeof value === "boolean") {
+      return value ? "True" : "False";
+    }
+
+    return value ? value : "No Data";
+  };
+
+  const columns = [
+    {
+      title: "Notice of Work #",
+      dataIndex: "now_number",
+      key: "now_number",
+    },
+    {
+      title: "Status",
+      key: "is_closed",
+      render: (record: any) => {
+        return record.is_closed ? "Closed" : "Open";
+      },
+    },
+    {
+      title: "Amendment",
+      key: "order_no",
+      dataIndex: "order_no",
+    },
+    {
+      title: "Changes",
+      dataIndex: "differences",
+      key: "differences",
+      render: (differences: IPermitDifference[]) => {
+        return (
+          <div className="padding-md--top">
+            {differences.map((diff) => (
+              <div key={diff.fieldName}>
+                {diff.fieldName === "Documents" ? (
+                  <div>
+                    <Typography.Paragraph strong className="margin-none line-height-none">
+                      Files Added:
+                    </Typography.Paragraph>
+                    {diff.currentValue.map((file: any, index) => (
+                      <Typography.Paragraph
+                        key={`${file}${index}`}
+                        className="green margin-none line-height-none"
+                      >
+                        {file}
+                      </Typography.Paragraph>
+                    ))}
+                  </div>
+                ) : (
+                  <div>
+                    <Typography.Paragraph strong className="margin-none line-height-none">
+                      {diff.fieldName}:
+                    </Typography.Paragraph>
+                    {diff.fieldName !== "None" && (
+                      <Typography.Paragraph>
+                        <Typography.Text className="red">
+                          {valueOrNoData(diff.previousValue)}
+                        </Typography.Text>
+                        {` => `}
+                        <Typography.Text className="green">
+                          {valueOrNoData(diff.currentValue)}
+                        </Typography.Text>
+                      </Typography.Paragraph>
+                    )}
+                  </div>
+                )}
+              </div>
+            ))}
+          </div>
+        );
+      },
+    },
+  ];
+
+  const data = Object.keys(currentDiff)
+    .map((key: any, index: number) => {
+      const amendment = explosivesPermit.explosives_permit_amendments.find(
+        (amendment) => amendment.explosives_permit_amendment_id == key
+      );
+
+      const permit = key === "0" ? explosivesPermit : amendment;
+
+      return {
+        ...permit,
+        differences: currentDiff[key].length > 0 ? currentDiff[key] : [{ fieldName: "None" }],
+        order_no: index,
```

Comment:
_If you wanted to_ you could get rid of order_no and just use amendment_no but that's pretty minor.